editor/src/clj/editor/pipeline/bob.clj

```diff
@@ -135,48 +134,57 @@
-        (map (fn [^Option option]
-               [(.getLongOpt option)
-                (not= -1 (.getArgs option))]))
-        (.getOptions (Bob/getCommandLineOptions))))
+(def ^:private long-option->option-config
+  (coll/pair-map-by #(.-longOpt ^Bob$CommandLineOption %) (Bob/getCommandLineOptions)))
 
 (defn- long-option? [x]
-  (contains? long-option->has-arg x))
+  (contains? long-option->option-config x))
 
 (defn- parse-options
   "Given bob options map, parses to a tuple of cli args and internal options
 
   Returns a tuple of 2 elements:
   - cli options: array of strings
   - internal options: either a map of Project options (string->string) or nil"
-  [options-map]
-  (let [{:keys [undefined cli internal]}
-        (util/group-into {} []
-                         (fn option-group [e]
-                           (let [k (key e)]
-                             (cond
-                               (keyword? k) :internal
-                               (long-option? k) :cli
-                               :else :undefined)))
-                         identity
-                         options-map)]
-    (when undefined
-      (throw (IllegalArgumentException. (format "Undefined bob option%s: %s"
-                                                (if (< 1 (count undefined)) "s" "")
-                                                (string/join ", " (map key undefined))))))
-    [(into []
-           (mapcat (fn [[k v]]
-                     (let [k-str (str "--" k)]
-                       (if (long-option->has-arg k)
-                         (if (vector? v)
-                           (eduction (mapcat (fn [v] [k-str (str v)])) v)
-                           [k-str (str v)])
-                         (if v [k-str] [])))))
-           cli)
-     (when internal
-       (coll/pair-map-by (comp name key) (comp str val) internal))]))
+  [^Path project-root options-map]
+  (letfn [(validate-type [^Bob$CommandLineOption config ^Class expected-class value]
+            (when-not (instance? expected-class value)
+              (throw (IllegalArgumentException. (str (.-longOpt config) " option is expected to be " (.getSimpleName expected-class) ", was " (some-> (class value) .getSimpleName)))))
+            value)
+          (stringify-value [^Bob$CommandLineOption config value]
+            (condp = (.-argType config)
+              Bob$CommandLineOption$ArgType/UNSPECIFIED (str value)
+              Bob$CommandLineOption$ArgType/ABS_OR_CWD_REL_PATH (str (.normalize (.resolve project-root (str (validate-type config String value)))))))]
+    (let [{:keys [undefined cli internal]}
+          (util/group-into {} []
+                           (fn option-group [e]
+                             (let [k (key e)]
+                               (cond
+                                 (keyword? k) :internal
+                                 (long-option? k) :cli
+                                 :else :undefined)))
+                           identity
```

Comment:
You can simply omit the `value-fn` argument to `group-into` if you want to use the values as-is.

Reply:
Good catch!